rpmio/macro.c

```diff
@@ -1197,17 +1214,106 @@ static void doTrace(MacroBuf mb, int chkexist, int negate,
 static const char *setNegateAndCheck(const char *str, int *pnegate, int *pchkexist) {
 
     *pnegate = 0;
-    *pchkexist = 0;
+    *pchkexist = CHK_NO;
     while ((*str == '?') || (*str == '!')) {
 	if (*str == '!')
 	    *pnegate = !*pnegate;
 	else
-	    (*pchkexist)++;
+	    *pchkexist = CHK_BASIC;
 	str++;
     }
     return str;
 }
 
+static void partsInit(macroPartition *parts)
+{
+    parts->f = parts->fe = NULL;
+    parts->g = parts->ge = NULL;
+    parts->h = parts->he = NULL;
+    parts->fn = parts->gn = parts->hn = 0;
+    return;
```

Comment:
In general, never initialize more than one variable on a line, it only makes it harder to read. In a case like this, just memset() to zero.

The return at end of void function is redundant.